### PR TITLE
feat(FEC-13114): add chapters as segments to seekbar

### PR DESCRIPTION
**Pull request by lianbenjamin**
### Description of the Changes

part of the Advanced Timeline feature.
this PR contains changes and new code for supporting segments on the seekbar, whenever there are chapters.

- extracting the progress & buffered element to a new comp named ProgressBar
- add new state `segments` to seekbar
- add new actions for updating the segments

**related PR-** https://github.com/kaltura/playkit-js-timeline/pull/11

Solves FEC-13114

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/components/progress-bar.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  /**
+   * render component
+   *
+   * @param {*} props - component props
+   * @returns {React$Element} - component
+   * @memberof ProgressBar
+   */
+  render(props: any): React$Element<any> {
+    const bufferedWidth = `${Math.round(props.getBufferedPercent())}%`;
+    const progressWidth = `${props.player.isLive() && props.player.isOnLiveEdge() ? 100 : (props.currentTime / props.duration) * 100}%`;
```
[the remark this answers is not in the document]
Comment (lianbenjamin) [reply src/components/progress-bar.js:38]: right

File: src/components/seekbar/seekbar.js [hunk 2]
```diff
@@ -593,12 +593,14 @@ class SeekBar extends Component {
...
             {this.renderTimeBubble()}
-            <div className={style.buffered} style={{width: bufferedWidth}} />
-            {props.dataLoaded ? (
-              <div className={style.progress} style={{width: progressWidth}}>
-                {props.adBreak ? undefined : <a className={style.scrubber} />}
+            <ProgressBar getBufferedPercent={() => this.getBufferedPercent} />
+            {props.adBreak ? undefined : (
+              <div
+                id={'scrubber-container'}
+                style={`transform: translateX(${(props.currentTime / props.duration) * this._seekBarElement?.clientWidth}px)`}>
```
[the remark this answers is not in the document]
Comment (lianbenjamin) [reply src/components/seekbar/seekbar.js]: done

**Review by SivanA-Kaltura — DISMISSED** [review]

**Review by semarche-kaltura — APPROVED** [review]

File: src/components/seekbar/_seekbar.scss [hunk 3]
```diff
@@ -40,7 +40,13 @@
...
     transition: height 200ms ease, transform 200ms ease;
-
+    &.chapters {
+      background-color: transparent;
+      .virtual-progress-indicator {
+        height: 8px;
+        transform: translateY(-2px);
+      }
+    }
     .progress {
```
Comment (JonathanTGold) [inline src/components/seekbar/_seekbar.scss:50]: this style should move to a new sass file inside progress bar folder and progress bar file should move to its own folder

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

File: src/components/seekbar/_seekbar.scss (hunk 3 above)
Comment (lianbenjamin) [reply src/components/seekbar/_seekbar.scss]: done

**Review by JonathanTGold — APPROVED** [review]

**Pull request merged**